File: ios/TYLAI/BodyMassBridge.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BodyMassBridge, NSObject)

RCT_EXTERN_METHOD(fetchBodyMassSamplesSinceDays:(nonnull NSNumber *)days
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/TYLAI/BodyMassBridge.swift
import Foundation
import React

/// Registered with JS via `BodyMassBridge.m` (`RCT_EXTERN_MODULE`).
@objc(BodyMassBridge)
class BodyMassBridge: NSObject {
  @objc static func requiresMainQueueSetup() -> Bool { true }

  @objc(fetchBodyMassSamplesSinceDays:resolver:rejecter:)
  func fetchBodyMassSamplesSinceDays(
    _ days: NSNumber,
    resolver resolve: @escaping RCTPromiseResolveBlock,
    rejecter reject: @escaping RCTPromiseRejectBlock
  ) {
    let dayCount = max(1, min(days.intValue, 365))
    Task { @MainActor in
      let manager = HealthKitManager()
      await manager.requestAuthorization()
      let samples = await manager.fetchBodyMassSamples(sinceDays: dayCount)
      resolve(samples)
    }
  }
}
